receiver/sqlqueryreceiver/README.md

```diff
@@ -37,7 +37,21 @@ The configuration supports the following top-level fields:
   - [snowflake](https://github.com/snowflakedb/gosnowflake) - `username[:password]@<account_identifier>/dbname/schemaname[?param1=value&...&paramN=valueN]`
   - [sqlserver](https://github.com/microsoft/go-mssqldb) - `sqlserver://username:user_password@localhost:1433?database=db_name`
   - [tds](https://github.com/thda/tds) - `tds://username:user_password@localhost:5000/db_name`
-
+- `datasource_config` (optional): Used in place of the `datasource` string. Contains separate values that are processed separately, and then concatenated into the connection string.
+  ```yaml
+  host: localhost
+  port: 5432
+  database: mydb
+  username: myuser
+  password: ${DB_PASSWORD}
+  additional_options:
+    sslmode: disable
+    application_name: myapp
+  ```
+  - The `username` will be properly escaped and securely handled when building the connection string. Special characters in passwords (such as #, @, %, etc.) are automatically URL-encoded to ensure proper connection string formatting.
+  - The `password` field supports sensitive value handling through environment variables or other secure value sources. The value will be properly escaped and securely handled when building the connection string.
+  - The `additional_options` map can contain additional driver-specific connection parameters. These will be properly escaped and appended to the connection string.
+  - For the `sqlserver` driver, an instance appended to the hostname (e.g. `hostname1/instance1`) will be parsed properly into this connection string: `sqlserver://username:password@host:port/instance`.
```

Comment:
As commented above, I think it would be helpful to change the format here to state which options are available, and then as a part of that we can include the special case information, such as encoding.